OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -238,15 +241,15 @@ public short MovementCostForCell(CPos cell)
 			if (!world.Map.Contains(cell))
 				return short.MaxValue;
 
-			return cellsCost[cell];
+			return cell.Layer == 0 ? cellsCost[cell] : customLayerCellsCost[cell.Layer][cell];
```

Comment:
Just thinking aloud: Would it maybe make more sense to use the new `customLayer*` caches for `cell.Layer == 0` as well and drop the old `cellsCost` and `blockingCache`? Or would that negatively impact perf on the default layer?